storage/innobase/fil/fil0fil.cc

```diff
@@ -2078,51 +2044,17 @@ fil_ibd_create(
 	goto err_exit;
 }
 
-/** Try to open a single-table tablespace and optionally check that the
-space id in it is correct. If this does not succeed, print an error message
-to the .err log. This function is used to open a tablespace when we start
-mysqld after the dictionary has been booted, and also in IMPORT TABLESPACE.
-
-NOTE that we assume this operation is used either at the database startup
-or under the protection of dict_sys.latch, so that two users cannot
-race here. This operation does not leave the file associated with the
-tablespace open, but closes it after we have looked at the space id in it.
-
-If the validate boolean is set, we read the first page of the file and
-check that the space id in the file is what we expect. We assume that
-this function runs much faster if no check is made, since accessing the
-file inode probably is much faster (the OS caches them) than accessing
-the first page of the file.  This boolean may be initially false, but if
-a remote tablespace is found it will be changed to true.
-
-If the fix_dict boolean is set, then it is safe to use an internal SQL
-statement to update the dictionary tables if they are incorrect.
-
-@param[in]	validate	0=maybe missing, 1=do not validate, 2=validate
-@param[in]	purpose		FIL_TYPE_TABLESPACE or FIL_TYPE_TEMPORARY
-@param[in]	id		tablespace ID
-@param[in]	flags		expected FSP_SPACE_FLAGS
-@param[in]	name		table name
-If file-per-table, it is the table name in the databasename/tablename format
-@param[in]	path_in		expected filepath, usually read from dictionary
-@param[out]	err		DB_SUCCESS or error code
-@return	tablespace
-@retval	NULL	if the tablespace could not be opened */
-fil_space_t*
-fil_ibd_open(
-	unsigned		validate,
-	fil_type_t		purpose,
-	ulint			id,
-	ulint			flags,
-	fil_space_t::name_type	name,
-	const char*		path_in,
-	dberr_t*		err)
+fil_space_t *fil_ibd_open(ulint id, ulint flags,
+                          fil_space_t::validate validate,
+                          fil_space_t::name_type name,
+                          const char *path_in, dberr_t *err) noexcept
 {
 	mysql_mutex_lock(&fil_system.mutex);
 	fil_space_t* space = fil_space_get_by_id(id);
 	mysql_mutex_unlock(&fil_system.mutex);
 	if (space) {
-		if (validate > 1 && !srv_read_only_mode) {
+		if (validate == fil_space_t::VALIDATE_SPACE_ID
+		    && !srv_read_only_mode) {
 			fsp_flags_try_adjust(space,
 					     flags & ~FSP_FLAGS_MEM_MASK);
 		}
```

Comment:
Should it not be `validate >= fil_space_t::VALIDATE_SPACE_ID`  ?

Reply:
Before this change introduced the 4th value `fil_space_t::VALIDATE_IMPORT` (3), the expression `validate > 1` only matched the case that is now covered by `fil_space_t::VALIDATE_SPACE_ID` (2). In `ALTER TABLE…IMPORT TABLESPACE`, this adjustment that had to be introduced in ab1e6fefd869242d962cb91a006f37bb9ad534a7 is being performed elsewhere:
```c++
dberr_t
AbstractCallback::init(
	os_offset_t		file_size,
	const buf_block_t*	block) UNIV_NOTHROW
{
	const page_t*		page = block->page.frame;

	m_space_flags = fsp_header_get_flags(page);
	if (!fil_space_t::is_valid_flags(m_space_flags, true)) {
		ulint cflags = fsp_flags_convert_from_101(m_space_flags);
		if (cflags == ULINT_UNDEFINED) {
			return DB_CORRUPTION;
		}
		m_space_flags = cflags;
	}
```